packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -747,12 +748,14 @@ class _DragTargetState<T extends Object> extends State<DragTarget<T>> {
     setState(() {
       _candidateAvatars.remove(avatar);
     });
-    widget.onAccept?.call(avatar.data! as T);
-    widget.onAcceptWithDetails?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
+    if (avatar.data != null)  {
+      widget.onAccept?.call(avatar.data! as T);
+      widget.onAcceptWithDetails?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
+    }
   }
 
   void didMove(_DragAvatar<Object> avatar) {
-    if (!mounted) {
+    if (!mounted || avatar.data == null) {
```

Comment:
Why this change here? Is it covered by a test?

Reply:
Same as in `didDrop`, `avatar.data` can be `null` so it cannot be blindly casted as `avatar.data! as T`

I added a test in [test: Add a test to verify onMove is not called when data is null](https://github.com/flutter/flutter/pull/133136/commits/8ad5dc832f977be1cbc6033d120fc1ec8f238c60) to cover this change